ethers-core/src/utils/mod.rs

```diff
@@ -432,18 +434,22 @@ pub fn parse_bytes32_string(bytes: &[u8; 32]) -> Result<&str, ConversionError> {
     Ok(std::str::from_utf8(&bytes[..length])?)
 }
 
-/// The default EIP-1559 fee estimator which is based on the work by [MyCrypto](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/services/ApiService/Gas/eip1559.ts)
+/// The default EIP-1559 fee estimator.
+///
+/// Based on a conservative estimate
+/// of 2x base fee + 1.2 x median priority fees.
+/// safe against 0 base and priority fee values.
+/// derived from tested designs:
+/// alloy
+/// metamask: https://github.com/MetaMask/core/blob/v143.0.0/packages/gas-fee-controller/src/fetchGasEstimatesViaEthFeeHistory/calculateGasFeeEstimatesForPriorityLevels.ts
 pub fn eip1559_default_estimator(base_fee_per_gas: U256, rewards: Vec<Vec<U256>>) -> (U256, U256) {
     let max_priority_fee_per_gas = std::cmp::max(
         estimate_priority_fee(rewards),
         U256::from(EIP1559_FEE_ESTIMATION_DEFAULT_PRIORITY_FEE),
     );
     let potential_max_fee = base_fee_surged(base_fee_per_gas);
-    let max_fee_per_gas = if max_priority_fee_per_gas > potential_max_fee {
-        max_priority_fee_per_gas + potential_max_fee
-    } else {
-        potential_max_fee
-    };
+    let max_fee_per_gas = max_priority_fee_per_gas + potential_max_fee;
```

Comment:
we should always be adding the priority fee and the base fee

Reply:
It makes sense.